### DISCUSSION TITLE
`Badly formatted Response from Server` race condition?

### DISCUSSION THREAD

**Discussion by jacksonludwig** [trpc/trpc#3437] [Q&A]
I am using TRPC with react native (expo mobile frontend, aws lambda backend with adapter). I noticed that there is an issue where the client is unable to process the server's response that occurs seemingly randomly.

When using multiple `useQuery` calls, the frequency of the issue increases and occurs almost 100% of the time. With just one `useQuery` call, the vast majority of the time there are no issues but it still does occur.

I have verified the server is actually responding correctly by manually calling the TRPC endpoint. The response does not seem to effect whether the `Badly formatted Response from Server` error occurs.

It is quite difficult to create a reproducible environment for this, but I will look into it.

**Comment by KATT** [comment]
Sounds like a an odd problem, my guess is that it will be with the infra somehow, but would definitely be a high priority to fix if it's something within trpc.

**Reply by KATT** [reply]
@jacksonludwig my guess it's something with the infra being overloaded and returning some error (maybe rate limit), this would in turn make tRPC complain b/c it's not a response shape that aligns to what we expect

**Reply by jacksonludwig** [reply]
Thanks for your input. I am still not sure what's wrong, but I am trying to make a repo for reproducing the issue.

I don't think it's a rate limiting issue since it only occurs when using the trpc client; triggering lambdas by other means has no issues.

**Reply by jacksonludwig** [reply]
@KATT I believe I've narrowed down the issue. It seems to occur only when making multiple queries to DIFFERENT endpoints/functions.

Using my reproduction repo I get the issue consistently (almost 100% of the time). I am pretty confident now the issue is with the client, unless I have the basic configuration incorrect.

Here is my reproduction https://github.com/jacksonludwig/trpc-repro, it will require launching resources into an AWS account unfortunately (I feel it's likely the same error occurs without using AWS unless the adapter is at fault).

**Reply by KATT** [reply]
Please make this an issue and we'll look at it ASAP!

**Reply by jacksonludwig** [reply]
thanks, I posted an issue https://github.com/trpc/trpc/issues/3443
